src/components/Notification.tsx

```diff
@@ -6,41 +9,61 @@ interface NotificationProps {
   color?: Color
   textColor?: Color
   margin?: string | number
+  justify?: 'start' | 'center' | 'end' | 'between'
+  onDismiss?: () => void
+  icon?: string
 }
 
 export function Notification(props: NotificationProps) {
   if (!props.message) return null
+  const [isDismissed, setDismissed] = useState(false)
+
+  useEffect(() => {
+    if (isDismissed && props.onDismiss) {
+      setTimeout(() => {
+        if (props.onDismiss) props.onDismiss()
```

Comment:
Is the additional if check in here to protect against dismount? Should we return a clearTimeout instead? Maybe a custom useTimeout effect to handle this for us would be nice since we have this pattern in a few places. It could go next to useInterval in time.ts

Reply:
The additional check is because the typescript compiler can't determine that onDismiss is valid, despite the first if.  I imagine it's because the function execution happens inside a callback, so there's no guarantee.  I kept the first if there because there's no reason to set the timer if there's no onDismiss.  But yeah, I'll look at converting this to a useTimeout hook.